pkg/util/proxyutil/reverse_proxy.go

```diff
@@ -79,11 +79,37 @@ func wrapDirector(d func(*http.Request)) func(req *http.Request) {
 	}
 }
 
+// deletedHeaders lists a number of headers that we don't want to
+// pass-through from the upstream when using a reverse proxy.
+//
+// These are related to the connection between Grafana and the proxy
+// or instructions that would alter how a browser will interact with
+// future requests to Grafana (such as enabling Strict Transport
+// Security)
+var deletedHeaders = []string{
```

Comment:
Some of these already deleted by `httputil.ReverseProxy`, see https://github.com/golang/go/blob/ada0eec8277449ecd6383c86bc2e5fe7e7058fc7/src/net/http/httputil/reverseproxy.go#L294-L304

We've basically copied this for the call resource handling since it behaves like an HTTP proxy as well: 
https://github.com/grafana/grafana/blob/9599e8003b3476dbc6629d43b60e5c9eeee45b85/pkg/plugins/manager/client/client.go#L276-L294

Reply:
Ah, I can remove the duplicates then :+1: It's really the STS that's been reported as a problem, but I noticed we copied over Server as well and added the ones that seemed reasonable and safe to me, good to know some of them are already handled.

Reply:
Q: Should we have just one list of headers and use it in the different places? Is there a case in which we should forward some header for a request but not for other?

Reply:
Probably yes, but maybe not now. Me and @wbrowne have discussed unifying plugin/datasource proxy and call resource handling by refactoring/moving plugin/datasource proxy into plugin management. Then having a common handling of all this logic make more sense. Probably needs some more discussions/thoughts though.

Reply:
Updated to remove the extraneous headers in 37e5bac.